controllers/component_build_controller.go

```diff
@@ -253,7 +251,7 @@ func (r *ComponentBuildReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{}, nil
 	}
 
-	pipelineRef, err := r.GetBuildPipelineFromComponentAnnotation(ctx, &component)
+	_, err = r.GetBuildPipelineFromComponentAnnotation(ctx, &component)
 	if err != nil {
 		log.Error(err, fmt.Sprintf("Failed to read %s annotation on component %s", defaultBuildPipelineAnnotation, component.Name), l.Action, l.ActionView)
```

Comment:
function returns this error in case fallback
		err := fmt.Errorf("missing or empty pipeline annotation: %s, will add default one to the component", 

so from logged error it is clear what is happening